frontend/packages/console-app/src/components/cloud-shell/AutoResizeTerminal.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import { Terminal as XTerminal } from 'xterm';
+import Measure from 'react-measure';
+import * as fit from 'xterm/lib/addons/fit/fit';
+import * as full from 'xterm/lib/addons/fullscreen/fullscreen';
+
+XTerminal.applyAddon(fit);
+XTerminal.applyAddon(full);
+
+type TerminalProps = {
+  onData: any;
+  className?: string;
+};
+
+type TerminalState = {
+  width: number;
+  height: number;
+};
```

Comment:
You shouldn't need width and height in state. These will be set immediately on xterm.

Reply:
removed state from component